internal/testrunner/runners/system/servicedeployer/factory.go

```diff
@@ -5,27 +5,66 @@
 package servicedeployer
 
 import (
-	"errors"
 	"os"
 	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/logger"
 )
 
+const devDeployDir = "_dev/deploy"
+
 var (
 	// ErrNotFound is returned when the appropriate service runner for a package
 	// cannot be found.
 	ErrNotFound = errors.New("unable to find service runner")
 )
 
-// Factory chooses the appropriate service runner for the given package, depending
-// on service configuration files defined in the package.
-func Factory(packageRootPath string) (ServiceDeployer, error) {
-	packageDevPath := filepath.Join(packageRootPath, "_dev")
+// FactoryOptions defines options used to create an instance of a service deployer.
+type FactoryOptions struct {
+	PackageRootPath    string
+	DataStreamRootPath string
+}
+
+// Factory chooses the appropriate service runner for the given data stream, depending
+// on service configuration files defined in the package or data stream.
+func Factory(options FactoryOptions) (ServiceDeployer, error) {
+	devDeployPath, err := findDevDeployPath(options)
+	if err != nil {
+		logger.Errorf("can't find _dev/deploy directory")
+		return nil, ErrNotFound
+	}
 
 	// Is the service defined using a docker compose configuration file?
-	dockerComposeYMLPath := filepath.Join(packageDevPath, "deploy", "docker", "docker-compose.yml")
+	dockerComposeYMLPath := filepath.Join(devDeployPath, "docker", "docker-compose.yml")
 	if _, err := os.Stat(dockerComposeYMLPath); err == nil {
 		return NewDockerComposeServiceDeployer(dockerComposeYMLPath)
 	}
 
+	// Is the service defined using Terraform definition files?
+	terraformDirPath := filepath.Join(devDeployPath, "tf")
+	if _, err := os.Stat(terraformDirPath); err == nil {
+		return NewTerraformServiceDeployer(terraformDirPath)
+	}
 	return nil, ErrNotFound
```

Comment:
Nit: please add newline to separate the final return statement from the two deployer cases.

Reply:
Added, but I'm wondering why is this preference here :) ?

Reply:
Total personal preference, I suppose :) To me the two cases seem to be their own separate logical blocks from everything around them. But this is one of those things you can totally ignore in the future. In general, feel free to ignore anything I prefix with `Nit: `.